packages/kuma-gui/src/app/connections/sources.ts

```diff
@@ -43,6 +43,20 @@ const filter = (data: Record<string, unknown>, cb: (key: string, arr: unknown[])
 }
 export const sources = (source: Source, api: KumaApi) => {
   return defineSources({
+    '/connections/stats/for/zone-ingress/:name/:socketAddress': async (params) => {
+      const { name, socketAddress } = params
+      const res = await api.getZoneIngressData({
+        zoneIngressName: name,
+        dataPath: 'stats',
+      })
+      const connections = ConnectionCollection.fromObject(Stat.fromCollection(res))
+      return {
+        inbounds: Object.fromEntries(Object.entries(connections.listener).filter(([key, _value]) => key.startsWith(socketAddress.replace(':', '_')))),
+        outbounds: connections.cluster,
+        $raw: res,
+        raw: res,
```

Comment:
Looks like this is duplicate? For the naming I think we decided for `$raw` 🙂

Reply:
Yeah I wasn't totally sure about this, I just copied how it was elsewhere to remain the same. Let me see if I can remember why I originally did both, I think it was a similar things to the fact that we have `config` and `$raw` only here its not `config` but i had to name it something so it ended up being `raw` (`$raw` was originally supposed to be a non-obvious to access property)

Reply:
Ah yeah, I just went back to look at this.

The idea of `$raw` is that it shouldn't be amended or used unless we absolutely need to. It should be the exact response from the endpoint and we should avoid using it unless we are in a sticky situation (maybe as a temporary escape hatch).

`/stats` is unlike most other endpoints in that its plain text. We put this context in `raw` in order to display it. We also then add parsed and reshaped JSON as sibling properties for a when we display it differently to plain text (using GUI cards).

Essentially, right now at least, `raw` is the same as `$raw`, but we should avoid displaying `$raw` or potentially mutating `$raw`.

Does that help here? Do you think that thinking makes sense? i.e. keeping a semi-private `$raw` as a reference to the original HTTP response?

Reply:
hmm 🤔 I don't really see a benefit in having both identical entries. Also, if `$raw` is not meant to be accessed/used anywhere, why should we add it then? If this is a preparation for a sticky situation we might or might not meet in the future, I would see it as a premature optimization.
However, I do see a benefit in having the exact response from the API somewhere next to the sibling entries that we add in the data layer. So, I would definitely keep one of them 🙂

Reply:
> Also, if $raw is not meant to be accessed/used anywhere, why should we add it then? If this is a preparation for a sticky situation we might or might not meet in the future, I would see it as a premature optimization.

We've had times in the past where we've mistakenly added to the actual HTTP response (by adding our own custom props), which have then ended up being displayed in config. `raw` is supposed to be the actual content we want to show to users (in other places with this called `config`, its not called that here because its not really "config"), importantly without the additional custom props.

It might not be super clear why we then have another additional `$raw` property. This is the content without anything added to it, but also without anything _removed_ from it i.e. the original HTTP response. We sometimes also remove things from the thing we show to users. If we remove it from `raw/config` and its not contained in any custom properties, then we've lost that information completely.

As `$raw` is the actual unedited response, it continues to contain that information everywhere, and aswell as being an escape hatch is also sometimes useful when debugging/console.logging so I guess that last point is an extra benefit aswell as the escape hatch.

Let me know if this is clear, completely get it's weird that its seem that there are two, but the TLDR is:

1. `raw` (sometimes `config`) contains the thing we want to show to the user. Without any custom props and with some non-custom props removed.
2. `$raw` (which I think we wanted to make consistent form a naming point of view), is the actual HTTP response. No custom props, and _still containing some things that we might remove from `raw`_.

Again, I'm glad you asked this as its helped my think about why we do this and write it down somewhere!

Apart from that, and have re-understood myself why we do this, I'm inclined to keep both, and continue to add `$raw` properties if we see anywhere where there currently isn't one. wdyt?

Reply:
> Let me know if this is clear, completely get it's weird that its seem that there are two, but the TLDR is:
>
> raw (sometimes config) contains the thing we want to show to the user. Without any custom props and with some non-custom props removed.
$raw (which I think we wanted to make consistent form a naming point of view), is the actual HTTP response. No custom props, and still containing some things that we might remove from raw.

Ok, I think I got your point. Probably it makes more sense to me when I face such a case in which I want to show a `config` or (`raw`) but remove something from it.
But, am I right that we are currently neither referencing `raw` nor `$raw`?

Reply:
> But, am I right that we are currently neither referencing `raw` nor `$raw`?

Oh yes totally correct! 🤦 oh I see why this is even more confusing now!

Umm, if you look at the dataplane equivalent of this, you'll see where we reference `raw` (not `$raw`) and we'll eventually do the same here for zones (the "entire proxy" `Stats` tab will need changing to use the new source (which exposes `raw`), and we we add the "per 'bound" drawer Stats tab will also use it.

Reply:
I see 🙂 so this is also a preparation for further additions/changes. I still think that it would be better to add things when they are actually needed, just for reference. But I guess that's just a nit.
So, lets move on then 😄